lib/explorer/polars_backend/series.ex

```diff
@@ -235,9 +235,11 @@ defmodule Explorer.PolarsBackend.Series do
   def divide(left, right) when is_number(right), do: divide(left, scalar_rhs(right, left))
 
   @impl true
-  def pow(left, exponent) when is_number(exponent),
+  def pow(left, exponent) when is_float(exponent),
     do: Shared.apply_native(left, :s_pow, [exponent])
 
+  def pow(left, exponent) when is_integer(exponent),
```

Comment:
```suggestion
  def pow(left, exponent) when is_integer(exponent) and exponent >= 0,
```

Reply:
I think this change would result in a exception if the user try to evaluate the following:

```Ex
iex> s = [2, 4, 6] |> Explorer.Series.from_list()
iex> Explorer.Series.pow(s, -3.0)
```